TodoApp/TodoApp.Services/Updaters/ItemUpdater.cs

```diff
@@ -11,15 +10,18 @@ namespace TodoApp.Services.Updaters
     public class ItemUpdater : IItemUpdater
     {
         private readonly IDateTimeGenerator _dateTimeGenerator;
+        private readonly IItemRepository _itemRepository;
 
-        public ItemUpdater(IDateTimeGenerator dateTimeGenerator)
-            => _dateTimeGenerator = dateTimeGenerator;
+        public ItemUpdater(IItemRepository itemRepository, IDateTimeGenerator dateTimeGenerator)
+        {
+            _itemRepository = itemRepository;
+            _dateTimeGenerator = dateTimeGenerator;
+        }
 
-        public async Task<Item> UpdateItemInCollection(IMongoCollection<Item> collection, Guid id, Item item)
+        public async Task<Item> UpdateItem(Guid id, Item item)
```

Comment:
This method should change everything that should be change in item. Therefore taking new and previous item as a property is neater way 😉